lib/Doctrine/Common/Util/ClassUtils.php

```diff
@@ -106,4 +106,20 @@ public static function generateProxyClassName($className, $proxyNamespace)
     {
         return rtrim($proxyNamespace, '\\') . '\\' . Proxy::MARKER . '\\' . ltrim($className, '\\');
     }
+
+    /**
+     * Checks whether provided class name is aliased class name or not.
+     *
+     * @param string $className
+     *
+     * @return bool
+     */
+    public static function isClassNameAliasedClassName($className)
+    {
+        if (strpos($className, ':') === false) {
+            return false;
+        }
+
+        return false === class_exists($className);
```

Comment:
This is not sufficient to define whether a class is aliased: `ClassUtils::isClassNameAliasedClassName('💩')` is unlikely a valid alias.

Reply:
It will not be detected as alias because it does not have a `:`. This PR's idea was to allow anonymous classes that have `:` in their FQN not to be treated as aliased class names.

Since I'm not that familiar with Doctrine internals, I tried to poke as little as possible and check whether a class name is real class (when there is `:` present) would mean that it's not aliased class